### PR TITLE
Adds a logger to the GRPC server and polishes a test client configuration

### PR DESCRIPTION
**Pull request by mdimjasevic**
This is a simple patch that:

* adds a logger to the GRPC server, and
* polishes a test client configuration.

### CODE REVIEW

File: lib/Fencer/Server.hs [hunk 1]
```diff
@@ -45,7 +45,7 @@ runServerWithPort (Port port) logger appState = do
     let options = Grpc.defaultServiceOptions
             { Grpc.serverHost = "0.0.0.0"
             , Grpc.serverPort = fromIntegral port
-              -- TODO: set the logger
+            , Grpc.logger     = \s -> Logger.info logger $ Logger.msg s
```
Comment (effectfully) [inline lib/Fencer/Server.hs]: I'd write `Grpc.logger     = Logger.info logger . Logger.msg`, but it's not important in any way.

File: lib/Fencer/Types.hs [hunk 2]
```diff
@@ -188,4 +188,4 @@ data RuleBranch = RuleBranch
 ----------------------------------------------------------------------------
 
 -- | A network port wrapper
-newtype Port = Port Word deriving newtype (Eq, Show)
+newtype Port = Port { getPort :: Word } deriving newtype (Eq, Show)
```
Comment (effectfully) [inline lib/Fencer/Types.hs]: There are two conventions on how to prefix newtype unwrappers: with `get` like you did (that's what `base` does) and with `un` like people often do in production. For me `getPort` means a function that somehow gets, you know, a `Port`, so I like the `un` convention. Just saying, it's not important.

**Review by effectfully — APPROVED** [review]

File: lib/Fencer/Server.hs (hunk 1 above)
Comment (mdimjasevic) [reply lib/Fencer/Server.hs]: You're right! I attempted with something almost identical, but I got a type error and reverted to this. Definitely adopting your suggestion.

File: lib/Fencer/Types.hs (hunk 2 above)
Comment (mdimjasevic) [reply lib/Fencer/Types.hs]: Thanks for the note! I adopted your proposal with `un`.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -56,12 +56,13 @@ integration:
     - docker network inspect test >/dev/null 2>&1 || docker network create test
 
     # Run Fencer
+    - CNAME="fencer-"$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 25 ; echo '')
```
Comment (neongreen) [inline .gitlab-ci.yml:59]: @mdimjasevic this should have a comment re/ why it was done.

Comment (neongreen) [reply .gitlab-ci.yml:59]: Alternatively, I think you can remove it altogether (and `docker network inspect test`), the CI issue was due to old GitLab runner.

**Pull request merged**